packages/stack/proxy/src/rpc-manager.ts

```diff
@@ -0,0 +1,193 @@
+        // get connection directly to the node
+        const provider = await this.events.request2("blockchain:node:provider", "ethereum");
+        this._web3 = new Web3(provider);
+      }
+      return this._web3;
+    })();
+  }
+
+  private get nodeAccounts() {
+    return (async () => {
+      if (!this._nodeAccounts) {
+        const web3 = await this.web3;
+        this._nodeAccounts = await web3.eth.getAccounts();
+      }
+      return this._nodeAccounts || [];
+    })();
+  }
+
+  private get accounts() {
+    return (async () => {
+      if (!this._accounts) {
+        const web3 = await this.web3;
+        const nodeAccounts = await this.nodeAccounts;
+        this._accounts = AccountParser.parseAccountsConfig(this.embark.config.blockchainConfig.accounts, web3, dappPath(), this.logger, nodeAccounts);
+      }
+      return this._accounts || [];
+    })();
+  }
+
+  public init() {
+    this.registerActions();
+    this.setCommandHandlers();
+  }
+
+  private registerActions() {
+    this.embark.registerActionForEvent("tests:config:updated", { priority: 40 }, (_params, cb) => {
+      // blockchain configs may have changed (ie endpoint)
+      this._web3 = null;
+
+      // web.eth.getAccounts may return a different value now
+      // update accounts across all modifiers
+      this.resetAccounts(cb);
+    });
+
+    this.embark.registerActionForEvent("blockchain:proxy:request", this.onProxyRequest.bind(this));
+    this.embark.registerActionForEvent("blockchain:proxy:response", this.onProxyResponse.bind(this));
+  }
+
+  private setCommandHandlers() {
+    this.events.setCommandHandler("rpc:accounts:reset", this.resetAccounts.bind(this));
+
+    this.events.setCommandHandler("rpc:request:interceptor:register", this.registerRequestInterceptor.bind(this));
+    this.events.setCommandHandler("rpc:response:interceptor:register", this.registerResponseInterceptor.bind(this));
+  }
+
+  private async onProxyRequest<T>(params: ProxyRequestParams<T>, callback: Callback<ProxyRequestParams<T>>) {
+    try {
+      params = await this.executeInterceptors(this.requestInterceptors, params);
+    } catch (err) {
+      err.message = __("Error executing RPC request modifications for '%s': %s", params?.request?.method, err.message);
+      return callback(err);
+    }
+    callback(null, params);
+  }
+
+  private async onProxyResponse<T, R>(params: ProxyResponseParams<T, R>, callback: Callback<ProxyResponseParams<T, R>>) {
+    try {
+      params = await this.executeInterceptors(this.responseInterceptors, params);
+    } catch (err) {
+      err.message = __("Error executing RPC response modifications for '%s': %s", params?.request?.method, err.message);
+      return callback(err);
+    }
+    callback(null, params);
+  }
+
+  private registerRequestInterceptor<T>(filter: string | RegExp, interceptor: RpcRequestInterceptor<T>, options: RegistrationOptions = { priority: 50 }, callback?: Callback<null>) {
```

Comment:
Don't forget to document the `{priority: number}` optional parameter in the registration APIs.